### PR TITLE
Handle kin errors gracefully in chs-delta-api validator

### PR DESCRIPTION
**Pull request by amoriartyCH**
We were encountering issues where the kin library was returning unknown errors back to the format method and we were type casting them to the wrong types resulting in Go panics happening.

Rework the handling of returned errors from kin lib:
Using the errors.As and errors.Is functions introduced in go
v1.13 we can now handle errors more gracefully instead of using
a type switch with potential to panic the application.

Added use of these new methods/functions and reworked private
methods in chOpenApiValidator.go to be easier to read and maintain

Add unit test to assert request body validation is functioning correctly

**Resolves**
DSND-115

### CODE REVIEW

File: validation/chOpenApiValidator.go [hunk 1]
```diff
@@ -146,42 +140,116 @@ func formatError(contextId string, err error) []byte {
...
+	// err is a *MultiError, and mea is set to the error's value, so range over it to grab each error.
+	for _, e := range *mea {
+		// Begin comparing the retrieved error (e) to possible returned error types.
 
-		// Cast to SchemaError so that we can pull out all of the necessary data to build our CH Errors response.
-		schemaError := me.(*openapi3.SchemaError)
-		reason := strings.Replace(schemaError.Reason, "\"", "'", -1)
-		jsonPath := strings.Join(schemaError.JSONPointer(), ".")
-		fieldName := schemaError.JSONPointer()[len(schemaError.JSONPointer())-1]
+		// If we have a request error, pass re into the extractRequestError func to get a formatted CHError back.
```
Comment (emead) [inline validation/chOpenApiValidator.go:149]: Should this read `into the handleRequestError`?

File: validation/chOpenApiValidator.go [hunk 2]
```diff
@@ -146,42 +140,116 @@ func formatError(contextId string, err error) []byte {
...
+	var mea openapi3.MultiError
+	if ok := errors.As(re.Err, &mea); ok {
+		return handleMultiError(contextId, &mea, errsArray)
+	}
 
-		errsArray = append(errsArray, err)
+	// If it isn't a MultiError then we can begin the extract the error contents straight away.
+	var se *openapi3.SchemaError
+	if ok := errors.As(re.Err, &se); ok {
+		errsArray = handleSchemaError(se, errsArray)
```
Comment (emead) [inline validation/chOpenApiValidator.go:183]: No need to pass `errsArray` into `handleSchemaError` and `handleParseError`.
These functions could just return an error which could then be appended to the array.

File: validation/chOpenApiValidator.go (hunk 1 above)
Comment (amoriartyCH) [reply validation/chOpenApiValidator.go]: Fixed this, good spot. Also found another comment with an outdated function name so fixed that too in latest commit.

File: validation/chOpenApiValidator.go (hunk 2 above)
Comment (amoriartyCH) [reply validation/chOpenApiValidator.go]: Changed both functions to just return an error which is appended onto the errsArray when calling.

**Review by emead — APPROVED** [review]

**Pull request merged**